brave/brave-core
# Update on-ramp providers and add purchase method screen

## Resolves https://github.com/brave/brave-browser/issues/23316 (Update on-ramp methods)
## Resolves https://github.com/brave/brave-browser/issues/27996 (Switch to new API `getBuyUrlV1`)

Implements a new screen to select a purchase method containing Ramp.Network, Sardine and Transak.
Adopted [new API `getBuyUrlV1`](https://github.com/brave/brave-core/blob/03673e48c89965d88856204a78d3e1365645dfe0/components/brave_wallet/browser/asset_ratio_service.cc#L180).

* ⚠️ Sardine will appear only in US locales.
* Ramp.Network needs modified symbols obtained by `mapToRampNetworkSymbol`.
* New core method that accepts an array of providers, and returns the full list.
* Implemented `removeDuplicates` to avoid showing duplication in the `EditVisibleAssetsBottomSheetDialogFragment` UI.
* Update Sardine with new logo.


<img width="515" alt="Screenshot 2023-01-31 at 1 29 34 PM" src="https://user-images.githubusercontent.com/3308503/215760233-4371b093-9324-416e-b14d-2a073a695f75.png">

<img width="515" alt="Screenshot 2023-01-31 at 1 31 30 PM" src="https://user-images.githubusercontent.com/3308503/215760537-13deae14-e668-4987-be5c-3eebeccd04cc.png">

## Demo

https://user-images.githubusercontent.com/3308503/215813132-85e15ad3-97ce-48e2-8e5b-1aa33fb52eef.mov

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Open the wallet section
- Tap on FAB icon and select "Buy"
- Tap on "Select purchase method" button
- Observe the new screen "Select purchase method"
- Verify that tapping on the "Buy with Ramp" button opens a new page on Ramp network
- Verify that tapping on the "Buy with Sardine" button opens a new page on Sardine
- Verify that tapping on the "Buy with Transak" button opens a new page on Transak
